web/angular-wallet/src/app/main/messages/messages.service.ts

```diff
@@ -142,7 +130,7 @@ export class MessagesService implements Resolve<any> {
     };
 
     let transactionId;
-    if (this.onOptionsSelected.value.encrypt) {
```

Comment:
this is wild coupling...should be avoided. We have more issues in this service. IMHO a service should have no deps at all. and used imperatively

Reply:
the options component needs to update other components...like the toolbar. That's right, but the service shouldn't be coupled to that. That's why I pass the encryption thing as parameter. And this works fine. Tested.